micro_manager/snapshot/dataset.py

```diff
@@ -0,0 +1,238 @@
+from importlib import metadata
+import os
+from datetime import datetime
+
+import numpy as np
+
+try:
+    import h5py
+except ImportError:
+    raise ImportError(
+        "The Micro Manager snapshot computation requires the h5py package."
+    )
+
+import numpy as np
+
+try:
+    import h5py
+except ImportError:
+    raise ImportError(
+        "The Micro Manager snapshot computation requires the h5py package."
+    )
+
+
+class ReadWriteHDF:
+    def __init__(self, logger) -> None:
+        self._logger = logger
+        self._has_datasets = False
+
+    def create_file(self, file_path: str) -> None:
+        """
+        Create an HDF5 file for a given file name and path.
+
+        Parameters
+        ----------
+        file_path : str
+            File name added to the path to the file.
+
+        """
+        f = h5py.File(file_path, "w")
+        f.attrs["status"] = "writing"
+        f.attrs["MicroManager_version"] = str(metadata.version("micro-manager-precice"))
+        f.attrs["date"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        f.close()
+
+    def collect_output_files(
+        self, dir_name: str, file_list: list, database_length: int
+    ) -> None:
+        """
+        Iterate over given HDF5 files in a given directory and copy the content into a single file.
+        The files are deleted after the content is copied.
+
+        Parameters
+        ----------
+        dir_name : str
+            Path to directory containing the files.
+        file_list : list
+            List of files to be combined.
+        dataset_length : int
+            Global number of snapshots.
+        """
+        # Create a output file
+        main_file = h5py.File(os.path.join(dir_name, "snapshot_data.hdf5"), "w")
+        main_file.attrs["status"] = "writing"
+        main_file.attrs["MicroManager_version"] = str(
+            metadata.version("micro-manager-precice")
+        )
```

Comment:
Does this pick the latest version of the package?

Reply:
It picks the currently installed package version. If changes are made locally it will show that.

Reply:
Okay good :+1: